poi-ooxml/src/main/java/org/apache/poi/xwpf/usermodel/XWPFDocument.java

```diff
@@ -912,6 +920,53 @@ private void insertElementIntoCell(XWPFTableCell tableCell, IBodyElement iBodyEl
         }
     }
 
+    private void insertIntoTables(IBodyElement iBodyElement, Deque<XmlObject> path, CTTbl ctTbl, List<XWPFTable> tables) {
+        for (XWPFTable xwpfTable : tables) {
+            if (ctTbl == xwpfTable.getCTTbl()) {
+                insertElementIntoTable(xwpfTable, iBodyElement, path);
+            }
+        }
+    }
+
+    private void insertIntoHeadersFooters(IBodyElement iBodyElement, CTHdrFtrImpl ctHdrFtrImpl) {
+        for (XWPFHeader header : headers) {
+            if (ctHdrFtrImpl == header._getHdrFtr()) {
+                insertIntoHeaderFooter(iBodyElement, header);
```

Comment:
same issue as above about break call when you match in the if and likewise in next for loop.